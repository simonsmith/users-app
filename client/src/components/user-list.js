import React from 'react';
import {User} from './user';

export function UserList({userIds, userEntities, roleEntities}) {
  if (!userIds.length) {
    return null;
  }
  return (
    <ul data-testid="user-list">
      {userIds.map(userId => {
        const userData = userEntities[userId];
        return (
          <li key={userId}>
            <User {...userData} roleEntities={roleEntities} />
          </li>
        );
      })}
    </ul>
  );
}
