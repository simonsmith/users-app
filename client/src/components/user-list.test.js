import React from 'react';
import {shallow} from 'enzyme';
import {UserList} from './user-list';
import {useShallowEqualSelector} from '../util/use-shallow-equal-selector';

jest.mock('../util/use-shallow-equal-selector');

test('renders users from the store', () => {
  useShallowEqualSelector
    .mockReturnValueOnce({
      entities: {
        123: {
          name: 'foo',
          role: 111,
        },
        456: {
          name: 'baz',
          role: 111,
        },
      },
      result: ['123', '456'],
    })
    .mockReturnValueOnce({
      entities: {
        111: {
          foo: 'bar',
        },
      },
    });
  expect(shallow(<UserList />)).toMatchSnapshot();
});
