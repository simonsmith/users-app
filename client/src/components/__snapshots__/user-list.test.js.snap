// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders users from the store 1`] = `
<ul>
  <li
    key="123"
  >
    <User
      name="foo"
      role={
        Object {
          "foo": "bar",
        }
      }
    />
  </li>
  <li
    key="456"
  >
    <User
      name="baz"
      role={
        Object {
          "foo": "bar",
        }
      }
    />
  </li>
</ul>
`;
